src/healthkit.cs

```diff
@@ -3513,71 +3585,87 @@ interface HKWorkoutSession : NSSecureCoding {
 		[NullAllowed, Export ("endDate")]
 		NSDate EndDate { get; }
 
+		[NoiOS]
+		[NoMacCatalyst]
 		[Deprecated (PlatformName.WatchOS, 3, 0, message: "Use HKWorkoutSession (HKHealthStore, HKWorkoutConfiguration, out NSError) instead.")]
 		[Export ("initWithActivityType:locationType:")]
 		NativeHandle Constructor (HKWorkoutActivityType activityType, HKWorkoutSessionLocationType locationType);
 
-		[Deprecated (PlatformName.WatchOS, 5, 0, message: "Use HKWorkoutSession (HKHealthStore, HKWorkoutConfiguration, out NSError) instead.")]
+		[NoiOS]
 		[NoMacCatalyst]
+		[Deprecated (PlatformName.WatchOS, 5, 0, message: "Use HKWorkoutSession (HKHealthStore, HKWorkoutConfiguration, out NSError) instead.")]
 		[Export ("initWithConfiguration:error:")]
 		NativeHandle Constructor (HKWorkoutConfiguration workoutConfiguration, out NSError error);
 
+		[NoiOS]
 		[Watch (5, 0)]
 		[NoMacCatalyst]
 		[Export ("initWithHealthStore:configuration:error:")]
 		NativeHandle Constructor (HKHealthStore healthStore, HKWorkoutConfiguration workoutConfiguration, [NullAllowed] out NSError error);
 
 		[Watch (5, 0)]
-		[NoMacCatalyst]
 		[Export ("prepare")]
 		void Prepare ();
 
 		[Watch (5, 0)]
-		[NoMacCatalyst]
 		[Export ("startActivityWithDate:")]
 		void StartActivity ([NullAllowed] NSDate date);
 
 		[Watch (5, 0)]
-		[NoMacCatalyst]
 		[Export ("stopActivityWithDate:")]
 		void StopActivity ([NullAllowed] NSDate date);
 
 		[Watch (5, 0)]
-		[NoMacCatalyst]
 		[Export ("end")]
 		void End ();
 
 		[Watch (5, 0)]
-		[NoMacCatalyst]
 		[Export ("pause")]
 		void Pause ();
 
 		[Watch (5, 0)]
-		[NoMacCatalyst]
 		[Export ("resume")]
 		void Resume ();
 
+		[NoiOS]
 		[Watch (5, 0)]
 		[NoMacCatalyst]
 		[Export ("associatedWorkoutBuilder")]
 		HKLiveWorkoutBuilder AssociatedWorkoutBuilder { get; }
 
-		[Watch (9, 0), NoTV, NoiOS, NoMacCatalyst]
+		[Watch (9, 0), NoTV]
 		[Export ("beginNewActivityWithConfiguration:date:metadata:")]
 		void BeginNewActivity (HKWorkoutConfiguration workoutConfiguration, NSDate date, [NullAllowed] NSDictionary<NSString, NSObject> metadata);
 
-		[Watch (9, 0), NoTV, NoiOS, NoMacCatalyst]
+		[Watch (9, 0), NoTV]
 		[Export ("endCurrentActivityOnDate:")]
 		void EndCurrentActivity (NSDate date);
 
-		[Watch (9, 0), NoTV, NoiOS, NoMacCatalyst]
+		[Watch (9, 0), NoTV]
 		[Export ("currentActivity", ArgumentSemantic.Copy)]
 		HKWorkoutActivity CurrentActivity { get; }
+
+		[Watch (10, 0), NoTV, Mac (14, 0)]
+		[Export ("type")]
+		HKWorkoutSessionType Type { get; }
+
+		[Watch (10, 0), NoTV, NoMacCatalyst, NoMac, iOS (17, 0)]
+		[Export ("sendDataToRemoteWorkoutSession:completion:")]
+		[Async]
+		void SendDataToRemoteWorkoutSession (NSData data, Action<bool, NSError> completion);
+
+		[Watch (10, 0), NoTV, NoMacCatalyst, NoMac, NoiOS]
+		[Export ("startMirroringToCompanionDeviceWithCompletion:")]
+		void StartMirroringToCompanionDevice (Action<bool, NSError> completion);
```

Comment:
I think this method should have the [Async] attribute similar to the method you added above
The thing that triggered me to check is the completion Action parameter.

I confirmed with the web docs [here](https://developer.apple.com/documentation/healthkit/hkworkoutsession/4165515-startmirroringtocompaniondevicew?language=objc) and you can scroll down to see the photo below where they call it in the sample code with the await keyword
<img width="983" alt="image" src="https://github.com/xamarin/xamarin-macios/assets/50846373/0bd3a7c7-6ffa-4d2b-84dd-1d0dc25f5ac6">

Reply:
@tj-devel709 yep that's what intro was complaining about as well, fixed!